hooks/tk-multi-publish2/basic/publish_session.py

```diff
@@ -0,0 +1,712 @@
+        """
+        Method called by the publisher to determine if an item is of any
+        interest to this plugin. Only items matching the filters defined via the
+        item_filters property will be presented to this method.
+
+        A publish task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+            - accepted: Indicates if the plugin is interested in this value at
+                all. Required.
+            - enabled: If True, the plugin will be enabled in the UI, otherwise
+                it will be disabled. Optional, True by default.
+            - visible: If True, the plugin will be visible in the UI, otherwise
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+
+        path = _session_path()
+
+        if not path:
+            # the session has not been saved before (no path determined).
+            # provide a save button. the session will need to be saved before
+            # validation will succeed.
+            self.logger.warn(
+                "The Maya session has not been saved.",
+                extra=_get_save_as_action()
+            )
+
+        self.logger.info(
+            "Maya '%s' plugin accepted the current Maya session." %
+            (self.name,)
+        )
+        return {
+            "accepted": True,
+            "checked": True
+        }
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+
+        publisher = self.parent
+        path = _session_path()
+
+        if not path:
+            # the session still requires saving. provide a save button.
+            # validation fails.
+            self.logger.error(
+                "The Maya session has not been saved.",
+                extra=_get_save_as_action()
+            )
+            return False
+
+        # ensure we have an updated project root
+        project_root = cmds.workspace(q=True, rootDirectory=True)
+        item.properties["project_root"] = project_root
+
+        # warn if no project root could be determined.
+        if not project_root:
```

Comment:
Is it still a problem if there is no project root when there are templates defined?

Reply:
Not sure. What was the issue before?

Reply:
I was thinking that in ZC where it is fine to run without a root this would still warn.

Reply:
I've just changed it to `info` instead of `warn`. That way it won't be intrusive. Reasonable?